drivers/mtd_sdcard/mtd_sdcard.c

```diff
@@ -241,7 +241,7 @@ const mtd_desc_t mtd_sdcard_driver = {
         .params = &sdcard_spi_params[n]     \
     };                                      \
                                             \
-    mtd_dev_t CONCAT(*mtd, m) = (mtd_dev_t *)&mtd_sdcard_dev ## n
+    XFA(mtd_dev_xfa, 0) mtd_dev_t CONCAT(*mtd, m) = (mtd_dev_t *)&mtd_sdcard_dev ## n
```

Comment:
Should this be
```C
XFA(mtd_dev_xfa, 0) mtd_dev_t CONCAT(*mtd_sdcard, m) = (mtd_dev_t *)&mtd_sdcard_dev ## n
```
?

Reply:
No, this instruction creates `mtd_n` variable by intention.

Reply:
OK, so mtd_emulated is just handled differently because it is not a hardware device
```C
    XFA(mtd_dev_xfa, 0) mtd_dev_t CONCAT(*mtd_emulated, n) = (mtd_dev_t *)&mtd_emulated_dev ## n
```

Reply:
`mtd_emulated` is intended as an alternative to hardware MTDs and for test applications only (see [`MTD_EMULATED_DEV` doc](https://doc.riot-os.org/group__drivers__mtd.html#gadd9c83e5bbb8813796af3b210feddea6))

It is used when either no hardware MTD is present (that is `MTD_0` and `mtd0` are not defined) or when hardware MTDs are present but `CONFIG_USE_HARDWARE_MTD` is not defined. In that case, the test application just uses `mtd_emulated_dev0` instead of `MTD_0` or `mtd0`.

Reply:
:thinking: Since `mtd_emulated` is intended to be used only alternatively to hardware MTDs, we could also use `mtd0` to `mtd<n>` instead of `mtd_emulated0` to `mtd_emulated<n>` for them. There shouldn't be any duplicate symbol conflict.

Reply:
No, this does not work because the board definitions with hardware MTDs define `mtd0` in any case, regardless of whether `mtd_emulated` is used or not. We have to leave it like this and the emulated devices have to be used directly via the variables `my_emulated_dev0` to `my_emulated_dev<n>`.